src/v/net/tests/connection_rate_test.cc

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2022 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "net/connection_rate.h"
+#include "seastarx.h"
+
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/sleep.hh>
+#include <seastar/core/when_all.hh>
+#include <seastar/net/inet_address.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+#include <boost/test/tools/old/interface.hpp>
+
+#include <chrono>
+#include <unordered_map>
+
+namespace {
+
+// In test we can get situation when one fiber succesfull go in
+// semaphore.wait(), but increase counter for next_second
+void check_rate(int64_t rate, int64_t prev_diff, int64_t max_rate) {
+    BOOST_CHECK(rate <= max_rate + prev_diff);
+}
+
+} // namespace
+
+SEASTAR_THREAD_TEST_CASE(rate_test) {
```

Comment:
For unit tests, it's ideal to avoid sleeps/wallclock time as much as possible.  seastar has a `manual_clock` class for this.  Maybe `connection_rate` could be templated on the clock type, and instantiate it with a manual clock for tests.